compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -429,14 +429,14 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                             return;
                         }
                         let trait_ref = trait_predicate.to_poly_trait_ref();
-
+                        let (self_ty, file) = self.tcx.short_ty_string(trait_ref.skip_binder().self_ty());
                         let (post_message, pre_message, type_def) = self
                             .get_parent_trait_ref(obligation.cause.code())
-                            .map(|(t, s)| {
+                            .map(|(_, s)| {
                                 (
-                                    format!(" in `{t}`"),
-                                    format!("within `{t}`, "),
-                                    s.map(|s| (format!("within this `{t}`"), s)),
+                                    format!(" in `{self_ty}`"),
+                                    format!("within `{self_ty}`, "),
```

Comment:
This change isn't quite correct. `self_ty` corresponds to the type of the trait ref, but we want the type of the parent trait ref here (which corresponds to `t`). Instead what you have to do do is call `short_ty_string` on `t`.

Reply:
how do i get a `Ty<'tcx>` out of a `String`?

Reply:
@Milo123459 change `fn get_parent_trait_ref` to return the `Ty<'_>` instead of a `String`.